community/codegen/src/main/java/org/neo4j/codegen/source/SourceCode.java

```diff
@@ -61,6 +63,24 @@ public String toString()
         {
             return "SourceCode";
         }
+
+        private String fullClasspath()
+        {
+            StringBuilder classpath = new StringBuilder( System.getProperty( "java.class.path" ) );
+
+            ClassLoader classLoader = SourceCode.class.getClassLoader();
+            if ( classLoader instanceof URLClassLoader )
+            {
+                URL[] classpathUrls = ((URLClassLoader) classLoader).getURLs();
+                for ( URL classpathUrl : classpathUrls )
+                {
+                    classpath.append( File.pathSeparator );
+                    classpath.append( classpathUrl.getPath() );
+                }
+            }
```

Comment:
Absence of `java.class.path` makes `CodeGenerationTest` fail while running from maven. This happens because `SourceCode.class.getClassLoader().getURLs()` includes only `surefirebooter.jar`. So `codegen/target` is missing and `java.class.path` brings it.

I agree that it is safer to canonicalise and de-duplicate paths.